src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -105,23 +105,40 @@ public function add(array $headers)
      * @param string|null $default The default value
      * @param bool        $first   Whether to return the first value or all header values
      *
-     * @return string|string[]|null The first header value or default value if $first is true, an array of values otherwise
+     * @return string|string[]|int|null The first header value or default value if $first is true, an array of values otherwise
      */
     public function get($key, $default = null, $first = true)
```

Comment:
we usually remove deprecated arguments from the signature and fetch the argument dynamically